owncloudApp/src/main/java/com/owncloud/android/providers/FileContentProvider.kt

```diff
@@ -192,15 +193,21 @@ class FileContentProvider(val executors: Executors = Executors()) : ContentProvi
         return newUri
     }
 
-    private fun insert(db: SQLiteDatabase, uri: Uri, values: ContentValues): Uri {
+    private fun insert(db: SQLiteDatabase, uri: Uri, values: ContentValues?): Uri {
         when (uriMatcher.match(uri)) {
             ROOT_DIRECTORY, SINGLE_FILE -> {
-                val remotePath = values.getAsString(ProviderTableMeta.FILE_PATH)
-                val accountName = values.getAsString(ProviderTableMeta.FILE_ACCOUNT_OWNER)
+                val remotePath = values?.getAsString(ProviderTableMeta.FILE_PATH)
+                val accountName = values?.getAsString(ProviderTableMeta.FILE_ACCOUNT_OWNER)
                 val projection =
                     arrayOf(ProviderTableMeta._ID, ProviderTableMeta.FILE_PATH, ProviderTableMeta.FILE_ACCOUNT_OWNER)
                 val where = "${ProviderTableMeta.FILE_PATH}=? AND ${ProviderTableMeta.FILE_ACCOUNT_OWNER}=?"
-                val whereArgs = arrayOf(remotePath, accountName)
+                val whereArgs = arrayOf<String>()
```

Comment:
Is it not possible to use `listOf` here?

Reply:
`query` requires an array of non nullable strings, so i implemented it that way